web/server/codechecker_server/api/authentication.py

```diff
@@ -167,6 +206,85 @@ def performLogin(self, auth_method, auth_string):
                     codechecker_api_shared.ttypes.ErrorCode.AUTH_DENIED,
                     msg)
 
+        elif auth_method == "oauth":
+            provider, url = auth_string.split("@")
```

Comment:
If a client logs in with the `Username:Password` login method, but with the username `github@admin`, they will be assumed to have passed OAuth authentication. Which means they will be successfully logged in without a valid password, as an arbitrary user. (This is another mix-up attack, but this time with the authentication methods on CodeChecker side.)

Reply:
fixed, now it uses a separate function for creating session and handling Oauth authentication (ignoring all other authentication checks including Username:Password as it was described)